platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ResolutionStrategyInternal.java

```diff
@@ -30,6 +30,14 @@ public interface ResolutionStrategyInternal extends ResolutionStrategy {
      */
     void discardStateRequiredForGraphResolution();
 
+    /**
+     * Sets whether or not any configuration resolution is final and the state required for resolution can be
+     * discarded.  Setting this to false implies that the configuration may be re-resolved again in the future.
+     *
+     * Defaults to true.
+     */
+    void setResolutionShouldBeFinal(boolean resolutionShouldBeFinal);
```

Comment:
Rename property to `keepStateRequiredForGraphResolution` to indicate it is related to the `discard...` function above.

Reply:
Done